test/optimizers/test_optimizer_aqgd.py

```diff
@@ -93,6 +94,43 @@ def test_int_values(self):
 
         self.assertAlmostEqual(result.eigenvalue.real, -1.857, places=3)
 
+    def test_max_grouped_evals(self):
+        """Tests max_grouped_evals parameter"""
+        # Test max_grouped_evals for an objective function that can be parallelized #
+        aqgd = AQGD(momentum=0.0, max_evals_grouped=2)
+
+        vqe = VQE(
+            self.estimator,
+            ansatz=RealAmplitudes(),
+            optimizer=aqgd,
+            gradient=self.gradient,
+        )
+        result = vqe.compute_minimum_eigenvalue(operator=self.qubit_op)
+
+        self.assertAlmostEqual(result.eigenvalue.real, -1.857, places=3)
+        aqgd.set_max_evals_grouped(1)
+        self.assertAlmostEqual(result.eigenvalue.real, -1.857, places=3)
```

Comment:
Can you make this bit where you use the same VQE, but with different evals grouped a subtest. Or perhaps better split this VQE test out from the test below. They are more independent tests and then this test could be parameterized with using `ddt` eg heres an example, https://github.com/qiskit-community/qiskit-algorithms/blob/9f85dbcee08a80ff82ba82bac2ae19e06b3237ee/test/optimizers/test_optimizers.py#L187

This way you could test for 1, 2 and 3 and `ddt' via, the decorator, will make 3 separate tests - you would see that when running as it adds a suffice to the test name for each it creates.

Having separate tests is better, whether by separate functions or subTest. When they are separate then if the test fails, or say in an earlier assert in the same function with subTest later testing is still run. So as it stands here maybe it fails with grouped of 2, would it pass with VQE one, and what about the custom object test where it checks the the point, would that have provided some info had it run. Having other parts, that are logically independent, run and knowing if they passed or failed may give extra information that may help identify the failure.

Reply:
Makes sense. I will do so.